bootstrap/cmd/bootstrap/app/kfctlClient.go

```diff
@@ -0,0 +1,74 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/ratelimit"
+	httptransport "github.com/go-kit/kit/transport/http"
+	"golang.org/x/time/rate"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// KfctlClient provides a client to the KfctlServer
+type KfctlClient struct {
+	createEndpoint endpoint.Endpoint
+}
+
+// NewKfctlClient returns a KfctlClient backed by an HTTP server living at the
+// remote instance.
+func NewKfctlClient(instance string) (KfctlService, error) {
+	// Quickly sanitize the instance string.
+	if !strings.HasPrefix(instance, "http") {
+		instance = "http://" + instance
+	}
+	u, err := url.Parse(instance)
+	if err != nil {
+		return nil, err
+	}
+
+	// We construct a single ratelimiter middleware, to limit the total outgoing
+	// QPS from this client to all methods on the remote instance. We also
+	// construct per-endpoint circuitbreaker middlewares to demonstrate how
+	// that's done, although they could easily be combined into a single breaker
+	// for the entire remote instance, too.
+	limiter := ratelimit.NewErroringLimiter(rate.NewLimiter(rate.Every(time.Second), 100))
```

Comment:
Should we have the limiter on router side that rate limit total request of each router?

Reply:
Can you explain. This is rate limiting requests from router instances to the statefulsets. Are you taking about rate limiting the incoming requests to the router? How would we do that without using something like ISTIO/knative?